From 51dd3c85e8b2f57f2d87b7b73c96e4a46a246ba4 Mon Sep 17 00:00:00 2001
From: Ram Muthiah <rammuthiah@google.com>
Date: Mon, 25 Nov 2019 23:00:43 -0800
Subject: [PATCH 590/830] ANDROID: gki_defconfig: =m's applied for virtio
 configs in arm64
X-NVConfidentiality: public

aligned =m virtual dev configs for arm64 and 86_64

Test: Treehugger
Bug: 143710295
Change-Id: I77c8b838bec94cf5ad3f0dc124f8269c2b6d6636
Signed-off-by: Ram Muthiah <rammuthiah@google.com>
---
 arch/arm64/configs/gki_defconfig | 26 ++++++++++++++------------
 arch/x86/configs/gki_defconfig   |  3 +++
 2 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 72276cc6e492..a7b49f0ec9db 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -189,9 +189,11 @@ CONFIG_NET_CLS_BPF=y
 CONFIG_NET_EMATCH=y
 CONFIG_NET_EMATCH_U32=y
 CONFIG_NET_CLS_ACT=y
-CONFIG_VSOCKETS=y
-CONFIG_VIRTIO_VSOCKETS=y
+CONFIG_VSOCKETS=m
+CONFIG_VIRTIO_VSOCKETS=m
 CONFIG_BPF_JIT=y
+CONFIG_CAN=m
+CONFIG_CAN_VCAN=m
 CONFIG_BT=y
 CONFIG_CFG80211=y
 # CONFIG_CFG80211_DEFAULT_PS is not set
@@ -211,7 +213,7 @@ CONFIG_ZRAM=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_SIZE=8192
-CONFIG_VIRTIO_BLK=y
+CONFIG_VIRTIO_BLK=m
 CONFIG_UID_SYS_STATS=y
 CONFIG_SCSI=y
 # CONFIG_SCSI_PROC_FS is not set
@@ -229,7 +231,7 @@ CONFIG_DM_VERITY_FEC=y
 CONFIG_DM_BOW=y
 CONFIG_NETDEVICES=y
 CONFIG_TUN=y
-CONFIG_VIRTIO_NET=y
+CONFIG_VIRTIO_NET=m
 # CONFIG_ETHERNET is not set
 CONFIG_PHYLIB=y
 CONFIG_PPP=y
@@ -263,7 +265,7 @@ CONFIG_USB_USBNET=y
 # CONFIG_WLAN_VENDOR_TI is not set
 # CONFIG_WLAN_VENDOR_ZYDAS is not set
 # CONFIG_WLAN_VENDOR_QUANTENNA is not set
-CONFIG_VIRT_WIFI=y
+CONFIG_VIRT_WIFI=m
 CONFIG_INPUT_EVDEV=y
 CONFIG_KEYBOARD_GPIO=y
 # CONFIG_INPUT_MOUSE is not set
@@ -281,9 +283,9 @@ CONFIG_SERIAL_OF_PLATFORM=m
 CONFIG_SERIAL_AMBA_PL011=y
 CONFIG_SERIAL_AMBA_PL011_CONSOLE=y
 CONFIG_SERIAL_DEV_BUS=y
-CONFIG_VIRTIO_CONSOLE=y
+CONFIG_VIRTIO_CONSOLE=m
 CONFIG_HW_RANDOM=y
-CONFIG_HW_RANDOM_VIRTIO=y
+CONFIG_HW_RANDOM_VIRTIO=m
 # CONFIG_HW_RANDOM_CAVIUM is not set
 # CONFIG_DEVPORT is not set
 # CONFIG_I2C_COMPAT is not set
@@ -307,7 +309,7 @@ CONFIG_MEDIA_CONTROLLER=y
 # CONFIG_VGA_ARB is not set
 CONFIG_DRM=y
 # CONFIG_DRM_FBDEV_EMULATION is not set
-CONFIG_DRM_VIRTIO_GPU=y
+CONFIG_DRM_VIRTIO_GPU=m
 CONFIG_BACKLIGHT_CLASS_DEVICE=y
 CONFIG_SOUND=y
 CONFIG_SND=y
@@ -348,14 +350,15 @@ CONFIG_LEDS_TRIGGERS=y
 CONFIG_EDAC=y
 CONFIG_RTC_CLASS=y
 # CONFIG_RTC_SYSTOHC is not set
+CONFIG_RTC_DRV_TEST=m
 CONFIG_RTC_DRV_PL030=y
 CONFIG_RTC_DRV_PL031=y
 CONFIG_DMADEVICES=y
 CONFIG_UIO=y
-CONFIG_VIRTIO_PCI=y
+CONFIG_VIRTIO_PCI=m
 # CONFIG_VIRTIO_PCI_LEGACY is not set
-CONFIG_VIRTIO_INPUT=y
-CONFIG_VIRTIO_MMIO=y
+CONFIG_VIRTIO_INPUT=m
+CONFIG_VIRTIO_MMIO=m
 CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_STAGING=y
 CONFIG_ASHMEM=y
@@ -459,7 +462,6 @@ CONFIG_CRYPTO_MD4=y
 CONFIG_CRYPTO_LZ4=y
 CONFIG_CRYPTO_ZSTD=y
 CONFIG_CRYPTO_ANSI_CPRNG=y
-CONFIG_CRYPTO_DEV_VIRTIO=y
 CONFIG_CRC_CCITT=y
 CONFIG_CRC8=y
 CONFIG_XZ_DEC=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index d7cf31394e34..e7b81b524ace 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -253,6 +253,7 @@ CONFIG_SERIAL_8250=y
 CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_OF_PLATFORM=m
 CONFIG_SERIAL_DEV_BUS=y
+CONFIG_VIRTIO_CONSOLE=m
 CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_VIRTIO=m
 # CONFIG_I2C_COMPAT is not set
@@ -297,6 +298,8 @@ CONFIG_USB_CONFIGFS_F_MIDI=y
 CONFIG_MMC=m
 # CONFIG_PWRSEQ_EMMC is not set
 # CONFIG_PWRSEQ_SIMPLE is not set
+CONFIG_MMC_SDHCI=m
+CONFIG_MMC_SDHCI_PLTFM=m
 CONFIG_NEW_LEDS=y
 CONFIG_LEDS_CLASS=y
 CONFIG_LEDS_TRIGGERS=y
-- 
2.17.1

